File: C_&C++/c/array/MultiD/2_pascalTriangle.c
#include<stdio.h>

int main(){
    int arr[7][7]={0};

    int row=2,col=0;

    arr[0][0]=arr[1][1]=arr[1][0]=1;

    while (row<=7)
    {
        arr[row][0]=1;
        for(col=1;col<row;col++){
            arr[row][col]=arr[row-1][col]+arr[row-1][col-1];

        }
        arr[row][col]=1;
        row++;
    }
    
    printf("Printing Pascal triangle : \n");

    for (int i=0;i<7;i++){
        printf("\n");
        for (int j=0;j<=i;j++){
            printf("\t %d",arr[i][j]);
        }
    }
    printf("\n");
}

File: LeetCode/ArrayAndStrings/Arrays/2_PascalTriangle_.py
from typing import List


def generate(numRows):
    arr=[[ None for m in range(k)] for k in range(1,numRows+1)]
    print(arr)
    arr[0][0]=1
    for i in range(1,numRows):
        arr[i][0]=1
        j=1
        while(j<i):
            arr[i][j]=arr[i-1][j]+arr[i-1][j-1]
            j+=1
        arr[i][j]=1
    return arr
print(generate(5))


class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        triangle=[[1]*(i+1) for i in range(numRows)]
        # print(triangle)
        
        for row in range(2,numRows):
            for col in range(1,row):
                triangle[row][col]=triangle[row-1][col-1]+triangle[row-1][col]
        
        return triangle
        
sol=Solution()
print(sol.generate(1))